markets/perps-market/contracts/modules/AsyncOrderModule.sol

```diff
@@ -71,6 +71,12 @@ contract AsyncOrderModule is IAsyncOrderModule {
             );
         }
 
+        // Check if there's a previous position open on this market, if not check if the account can open a new position
+        if (PerpsMarket.accountPosition(commitment.marketId, commitment.accountId).size == 0) {
+            //is a new position, check if not exceeding max positions per account
+            PerpsAccount.load(commitment.accountId).canOpenNewPosition();
+        }
```

Comment:
i would move the if condition into the validation itself.  maybe something like this:

```
PerpsAccount.validateMaxPositions(commitment.marketId, commitment.accountId);
```

and load the position and account in the validation itself.  just cleans up the modules a bit more.

Reply:
wish we still had `createValid`, then this call for validating max positions can just live there.

i have some ideas on this, but can leave it to a separate PR 👍🏽